addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -293,14 +299,22 @@ def _get_bom_data(self, bom, warehouse, product=False, line_qty=False, bom_line=
         bom_report_line.update(availabilities)
 
         if level == 0:
-            if bom_report_line['producible_qty'] > 0:
-                bom_report_line['status'] = _("%(qty)s Ready To Produce", qty=bom_report_line['producible_qty'])
-            else:
-                bom_report_line['status'] = _("No Ready To Produce")
-        elif missing_qty := max(bom_report_line['quantity'] - bom_report_line['quantity_available'], 0):
-            missing_qty = float_repr(missing_qty, self.env['decimal.precision'].precision_get('Product Unit'))
-            route_name = bom_report_line['route_name'] or _('Order')
-            bom_report_line['status'] = _("%(qty)s To %(route)s", qty=missing_qty, route=route_name)
+            # Gives a unique key for the first line that indicates if product is ready for production right now.
+            bom_report_line['components_available'] = all(c['stock_avail_state'] == 'available' for c in components)
+            if bom_report_line['producible_qty'] > 0 and bom_report_line['components_available']:
+                bom_report_line['status'] = self.env._("%(qty)s Ready To Produce", qty=self._format_number_display(bom_report_line['producible_qty']))
+            elif bom_report_line['phantom_bom'] and not all(c['availability_state'] == 'available' for c in components):
+                bom_report_line['availability_state'] = "estimated"
+                bom_report_line['status'] = self._format_date_display(bom_report_line['availability_state'], bom_report_line['availability_delay'])
+        elif missing_qty := max(bom_report_line['quantity'] - bom_report_line['quantity_available'], 0) and bom_report_line['route_name'] == 'Order':  # Only kits use this route
```

Comment:
```suggestion
        elif missing_qty := max(bom_report_line['quantity'] - bom_report_line['quantity_available'], 0) and bom_report_line['phantom_bom']:
```
Keep things simple, you already have something that tells you if the bom is a kit or not.